### PR TITLE
Sass 430 2

### PR DESCRIPTION
**Pull request by AdamSmithHMRC**
Made to replace SASS-430

### CODE REVIEW

**Comment by juamalBlackman** [comment]
Tests are failing where tax year the tax year is not equal 2022

File: app/controllers/predicates/TaxYearAction.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  lazy val logger: Logger = Logger.apply(this.getClass)
+  implicit val config: AppConfig = appConfig
+  implicit val messagesApi: MessagesApi = messages
+
+  override def refine[A](request: User[A]): Future[Either[Result, User[A]]] = {
+    Future.successful(
+      if (taxYear == appConfig.defaultTaxYear || !appConfig.taxYearErrorFeature) {
+        Right(request)
+      } else {
+        Left(Redirect(controllers.routes.TaxYearErrorController.show()).addingToSession(SessionValues.TAX_YEAR -> config.defaultTaxYear.toString)(request))
```
Comment (juamalBlackman) [inline app/controllers/predicates/TaxYearAction.scala:45]: may aswell move addingTosession onto a newline for readability

File: app/controllers/predicates/TaxYearAction.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class TaxYearAction @Inject()(taxYear: Int)(
+  appConfig: AppConfig,
+  val messages: MessagesApi
+) extends ActionRefiner[User, User] with I18nSupport {
+
+  implicit val executionContext: ExecutionContext = ExecutionContext.global
+  lazy val logger: Logger = Logger.apply(this.getClass)
```
Comment (juamalBlackman) [inline app/controllers/predicates/TaxYearAction.scala:36]: nothing is being logged no longer needed

**Review by juamalBlackman — CHANGES_REQUESTED** [review]

File: app/controllers/predicates/TaxYearAction.scala (hunk 2 above)
Comment (BenCosfordHMRC) [reply app/controllers/predicates/TaxYearAction.scala:36]: we should copy the log that was there before

**Pull request merged**